rasa/core/actions/action.py

```diff
@@ -168,6 +175,59 @@ def __str__(self) -> Text:
         return "Action('{}')".format(self.name())
 
 
+class ActionUtterPredictedResponse(Action):
+    """An action which queries the Response Selector for the appropriate response."""
+
+    def __init__(self, name: Text, silent_fail: Optional[bool] = False):
+        self.action_name = name
+        self.silent_fail = silent_fail
+
+    async def run(
+        self,
+        output_channel: "OutputChannel",
+        nlg: "NaturalLanguageGenerator",
+        tracker: "DialogueStateTracker",
+        domain: "Domain",
+    ):
+        """Query the appropriate response and create a bot utterance with that."""
+
+        message = None
+
+        # Build query keys for tracker. This is a list because response for the action can be
+        # predicted by one of the two -
+        # 1. a specific response selector of this open domain intent type
+        # 2. a generic response selector built for all open domain intent types in the training data.
+        # We check in both
+        response_selector_keys = [
```

Comment:
do we expect more than two keys in here? if not I'd rather go for smth more readable along the lines of
```python
if DEFAULT_OPEN_UTTERANCE_TYPE_KEY in tracker.latest_message.parse_data:
  query_key = DEFAULT_OPEN_UTTERANCE_TYPE_KEY
elif OTHER_KEY in tracker.latest_message.parse_data:
  query_key = OTHER_KEY
else:
  if not self.silent_fail:
    logger.error(
      "Couldn't create message for response action '{}'."
      "".format(self.action_name)
    )
  return []

logger.debug("Picking response of type {}".format(query_key))
message = {
  "text": tracker.latest_message.parse_data[query_key].get("name")
}
return [create_bot_utterance(message)]
```

Reply:
We don't, there can always be atmost two valid corresponding keys